Remove salt patches

stefanc18 opened the pull request:
The patches are being applied upstream:
- https://github.com/ni/salt/pull/54
- https://github.com/ni/salt/pull/55
- https://github.com/ni/salt/pull/56

There is still one patch left which wasn't removed because we haven't added it upstream it. We will after those PRs from above get in, since it is dependent on them. Will follow up to remove the latest patch as soon as we have ready the PR to push that one upstream as well.

amstewart commented:
We use commits as the atomic unit of change in the OE layers. Could you squash your commits and make sure the commit message is meaningful and follows the styling of the other commits in the layer? And also sign your commits with `git commit -s`.

stefanc18 commented:
@amstewart I updated the commit, let me know if it is okay now

amstewart commented:
It still needs some commentary about why the patches are being removed, and needs your sign-off.

stefanc18 committed:
salt: remove 7 of the patches

Remove 7 of the patches that were solving compatibility issues with Python 3.10.
The reason they are being removed is that they are being applied in the repo upstream.

Signed-off-by: Ștefan Ciuprina <stefan.ciuprina@ni.com>
